Framework/PythonInterface/plugins/algorithms/SaveGEMMAUDParamFile.py

```diff
@@ -0,0 +1,204 @@
+                                              func_1_sigma_zeros=self._format_param_list(sigma_zeros),
+                                              func_1_sigma_ones=self._format_param_list(sigma_ones),
+                                              func_1_sigma_twos=self._format_param_list(sigma_twos),
+                                              func_2_alpha_zeros=create_empty_param_list(),
+                                              func_2_alpha_ones=create_empty_param_list(),
+                                              func_2_betas=create_empty_param_list(),
+                                              func_2_switches=create_empty_param_list(),
+                                              func_2_sigma_zeros=create_empty_param_list(),
+                                              func_2_sigma_ones=create_empty_param_list(),
+                                              func_2_sigma_twos=create_empty_param_list(),
+                                              func_2_gamma_zeros=create_empty_param_list(),
+                                              func_2_gamma_ones=create_empty_param_list(),
+                                              func_2_gamma_twos=create_empty_param_list(),
+                                              function_types=create_empty_param_list("1")
+                                              ))
+
+    def validateInputs(self):
+        issues = {}
+
+        input_ws = mtd[self.getPropertyValue(self.PROP_INPUT_WS)]
+        grouping_scheme = self.getProperty(self.PROP_GROUPING_SCHEME).value
+        if len(grouping_scheme) != input_ws.getNumberOfEntries():
+            issues[self.PROP_GROUPING_SCHEME] = ("Number of entries in {} does not match number of spectra in {}. "
+                                                 "You must assign a bank to every focused spectrum in the input workspace".
+                                                 format(self.PROP_GROUPING_SCHEME, self.PROP_INPUT_WS))
+
+        return issues
+
+    def _expand_to_texture_bank(self, bank_param_list, spectrum_numbers, grouping_scheme):
+        """
+        :param bank_param_list: a list of n values for some parameter, such as DIFC
+        :param spectrum_numbers: a list of m bank IDs, one for each focused spectrum
+        :param grouping_scheme: a list of m indexes, to index bank_param_list, where
+        the element at item i is the bank number in bank_param_list to associate
+        focused spectrum i with
+        :return: a list of m values, where each value is the assigned parameter for the
+        corresponding bank
+        """
+        return (bank_param_list[grouping_scheme[spec_num] - 1] for spec_num in spectrum_numbers)
+
+    def _find_isis_powder_dir(self):
+        script_dirs = config["pythonscripts.directories"].split(";")
+        diffraction_dir = None
+        for dir in script_dirs:
+            if "Diffraction" in dir:
+                diffraction_dir = dir
```

Comment:
Could we just use `filter` here and take the last one?

Reply:
```python
script_dirs = config["pythonscripts.directories"].split(";")
script_dirs = filter(lambda directory: "Diffraction" in directory, script_dirs)
diffraction_dir = script_dirs.pop() if len(script_dirs) > 0 else None
```

Reply:
Are we sure we always want the last directory?

Reply:
Good point - I've updated it to use the first one for which the template file actually exists